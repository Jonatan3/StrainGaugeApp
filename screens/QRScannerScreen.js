import React, { useState } from 'react'
import QRCodeScanner from 'react-native-qrcode-scanner'
import { RNCamera } from 'react-native-camera'
import { View, ToastAndroid, Dimensions } from 'react-native'
import StreamingConstructor from '../Streaming/StreamingConstructor'
import QRScannerOverlay from '../components/QRScannerOverlay'

export default function QRCodeScannerScreen({ navigation }) {
  const [flash, setFlash] = useState(RNCamera.Constants.FlashMode.off)

  const onFlashClick = () => {
    if (flash === RNCamera.Constants.FlashMode.torch) {
      setFlash(RNCamera.Constants.FlashMode.off)
    }
    else {
      setFlash(RNCamera.Constants.FlashMode.torch)
    }
  }

  const onSucces = (e) => {
    if (StreamingConstructor.getInstance().getChannelById(parseInt(e.data)) !== null) {
      //setTimeout(() => { this.scanner.reactivate() }, 100)
      navigation.push('Strain Gauge', { itemId: parseInt(e.data) })
    }
    else {
      ToastAndroid.show('ERROR! no strain gauge with id of ' + e.data + ' found!', ToastAndroid.LONG)
      //setTimeout(() => { this.scanner.reactivate() }, 3000)
    }
  }

  return (
    <View>
      <QRCodeScanner
        flashMode={flash}
        showMarker={true}
        onRead={(e) => onSucces(e)}
        cameraStyle={{ height: Dimensions.get('window').height }}
        reactivate={true}
        reactivateTimeout={1000}
        customMarker={
          <QRScannerOverlay onFlashClick={onFlashClick} flash={flash} navigation={navigation} />
        }
      />
    </View>
  )
}