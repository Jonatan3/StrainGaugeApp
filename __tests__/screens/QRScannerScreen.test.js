import React from 'react';
import renderer from 'react-test-renderer';
import QRScannerScreen from '../../screens/QRScannerScreen'

test('renders correctly', () => {
    const tree = renderer.create(<QRScannerScreen />).toJSON()
    expect(tree).toMatchSnapshot()
})