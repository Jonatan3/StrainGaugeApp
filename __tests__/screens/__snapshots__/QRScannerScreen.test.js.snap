// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 2,
            "justifyContent": "center",
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "height": 1334,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          ...
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 2,
            "justifyContent": "center",
            "width": 750,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
